src/cmap/auth/providers.ts

```diff
@@ -4,6 +4,7 @@ export const AuthMechanism = Object.freeze({
   MONGODB_CR: 'MONGODB-CR',
   MONGODB_DEFAULT: 'DEFAULT',
   MONGODB_GSSAPI: 'GSSAPI',
+  MONGODB_OIDC: 'MONGODB-OIDC',
```

Comment:
adding `@internal` here should make it not exist in the types (it will secretly be present at runtime 🤫 but I think that's okay)

Reply:
Done.